src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
Great work on your implementation of `calculateWomenAverageAge`. You've correctly used the `filter` method to select the women from the `people` array, and also handled the condition when `withChildren` is true. 

However, a few improvements could make your code cleaner and more efficient:

- Right now, you are filtering the women twice based on the `withChildren` parameter. This operation can be consolidated into a single `filter` call, which improves performance by reducing the number of iterations over the array.
- To check if a woman has children, you've used `find`, which will iterate over the entire array even after it has found a match. Instead, use `some`, which stops as soon as it finds a match, resulting in better performance.

```suggestion
  const women = people.filter(
    (p) => p.sex === 'f' && (!withChildren || people.some((child) => child.mother === p.name))
  );
```